src/app/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Component,
+  ElementRef,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import {CarouselItem} from '../../../carouselItem';
+import {carouselItems} from '../../../carouselItems';
+
+
+import {from, Observable } from 'rxjs';
+import { map } from 'rxjs/operators'
+
+@Component({
+  selector: 'app-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss'],
+})
+export class CarouselComponent implements OnInit{
+
+  private selectedSlideIndex = 0;
+  private translateX = 0;
+  carouselItems$!: Observable<CarouselItem>;
```

Comment:
Notice that it's an observable that emits **single** items. Example:

```--------xxxxxx``` where each x is an item from the mock data array;

`Observable<CarouselItem[]>` fits your case